Backpack/Chip/Classes/BPKChip.m

```diff
@@ -0,0 +1,192 @@
+    [self.layer addSublayer:self.tintLayer];
+
+    self.titleLabel = [[BPKLabel alloc] initWithFrame:CGRectZero];
+    self.titleLabel.fontStyle = BPKFontStyleTextSm;
+    [self addSubview:self.titleLabel];
+
+    self.titleLabel.textAlignment = NSTextAlignmentCenter;
+
+    [self addTarget:self action:@selector(handleSingleTap:) forControlEvents:UIControlEventTouchUpInside];
+
+    [self setupConstraints];
+
+    BPKShadow *shadow = [BPKShadow shadowSm];
+    [shadow applyToLayer:self.layer];
+
+    [self updateStyle];
+    self.initializing = NO;
+}
+
+#pragma mark - Layout overrides
+
+- (void)layoutSubviews {
+    self.tintLayer.frame = self.bounds;
+    self.tintLayer.cornerRadius = self.bounds.size.height / 2;
+
+    self.layer.cornerRadius = self.bounds.size.height / 2;
+}
+
+#pragma mark - State setters
+
+- (void)setTitle:(NSString *_Nullable)title {
+    _title = [title copy];
+
+    [self updateTitle];
+}
+
+- (void)setSelected:(BOOL)selected {
+    [super setSelected:selected];
+    [self updateStyle];
+}
+
+- (void)setHighlighted:(BOOL)highlighted {
+    [super setHighlighted:highlighted];
+    [self updateStyle];
+}
+
+- (void)setEnabled:(BOOL)enabled {
+    [super setEnabled:enabled];
+    [self updateStyle];
+}
+
+#pragma mark - Layout
+
+- (void)setupConstraints {
+    CGFloat chipHorizontalSpacing = [self chipHorizontalSpacing];
+    CGFloat chipVerticalSpacing = [self chipVerticalSpacing];
+
+    self.titleLabel.translatesAutoresizingMaskIntoConstraints = NO;
+
+    [NSLayoutConstraint activateConstraints:
+     @[
+       [self.titleLabel.leadingAnchor constraintEqualToAnchor:self.leadingAnchor constant:chipHorizontalSpacing],
+       [self.titleLabel.topAnchor constraintEqualToAnchor:self.topAnchor constant:chipVerticalSpacing],
+       [self.trailingAnchor constraintEqualToAnchor:self.titleLabel.trailingAnchor constant:chipHorizontalSpacing],
+       [self.bottomAnchor constraintEqualToAnchor:self.titleLabel.bottomAnchor constant:chipVerticalSpacing]
+       ]
+     ];
+}
+
+#pragma mark - Updates
+
+- (void)updateStyle {
+    if(self.selected) {
+        self.backgroundColor = BPKColor.blue500;
+        self.textColor = BPKColor.white;
+    } else {
+        self.backgroundColor = BPKColor.white;
+        self.textColor = BPKColor.gray700;
+    }
+
+    if(!self.enabled){
+        self.backgroundColor = BPKColor.white;
+        self.textColor = BPKColor.gray100;
+    }
+
+    if(self.highlighted){
+        [self animateTintLayer:YES];
+    }else{
+        [self animateTintLayer:NO];
+    }
+
+    [self updateTitle];
+}
+
+- (void)updateTitle {
+    if(self.title == nil){
+        return;
+    }
+
+    self.titleLabel.text = self.title;
+    [self.titleLabel setTextColor:self.textColor];
+}
+
+#pragma mark - Actions
+
+- (void)handleSingleTap:(UITapGestureRecognizer *)recognizer {
+    [self setSelected:!self.selected];
+}
+
+#pragma mark - Helpers
+
+- (void)animateTintLayer:(BOOL) tint {
```

Comment:
Don't use a space after the parenthesis here

Reply:
This method could do with a better name. I read `animateTintLayer:(BOOL)` as the bool controlling whether animation was used or not.

Reply:
Actually - it doesn't need a name at all.
See commit `discussion_r232996079`